java/code/src/com/redhat/rhn/domain/rhnset/RhnSetFactory.java

```diff
@@ -184,22 +184,19 @@ private static void executeMode(RhnSetElement elem,
         params.put("user_id", elem.getUserId());
         params.put("label", elem.getLabel());
         params.put("el_one", elem.getElement());
-        int count;
 
         if (elem.getElementThree() == null && elem.getElementTwo() == null) {
-            count = el1.executeUpdate(params);
+            el1.executeUpdate(params);
         }
         else if (elem.getElementThree() == null) {
             params.put("el_two", elem.getElementTwo());
-            count = el2.executeUpdate(params);
+            el2.executeUpdate(params);
         }
         else {
             params.put("el_three", elem.getElementThree());
             params.put("el_two", elem.getElementTwo());
-            count = el3.executeUpdate(params);
+            el3.executeUpdate(params);
         }
-
-        assert count == 1 : "Failed to update row";
```

Comment:
Fine to remove the assert, but... it does mean we are not checking the output of executeUpdate method?
What if the row failed updating it? :thinking:

Reply:
We do not have much to check. The update is restricted to just one row so the only possible outputs are two: 1 (row updated) or 0 (rows updated) - there cannot be more.

Before commit bb406bc, an output of 0 would signal an error, as we were definitely expecting 1 row to be updated. After the commit, we changed the query and both 1 and 0 are OK as outputs.

So there is nothing really left to check here, and in particular, failing on 0 is wrong, thus I am removing the check.